test/require-array-join-separator.mjs

```diff
@@ -71,5 +70,8 @@ test.snapshot({
 					)/**/
 			)
 		`,
+		'foo?.join()',
+		'foo.join?.()',
```

Comment:
I don't think this should be invalid. `String#join` is never optional, so if it is, it's a sign that it's a custom method and we should ignore it.

Reply:
You are right. I'll fix.